core/templates/dev/head/pages/exploration_editor/editor_tab/StateInteraction.js

```diff
@@ -362,7 +362,7 @@ oppia.controller('StateInteraction', [
         var imageWithRegions = currentCustomizationArgs.imageAndRegions.value;
         for (var j = 0; j < imageWithRegions.labeledRegions.length; j++) {
           _answerChoices.push({
-            val: imageWithRegions.labeledRegions[j].label,
+            val: j,
```

Comment:
I'm not sure about this. Doesn't it change what gets stored in the answer group in the backend? If so, this will break all the previous explorations.

Instead, I think the HTML-select directive has an error. What if we change the angular-html-bind in line 3 to `angular-html-bind="options[getSelectionIndex()].val` and introduce a new function as follows? Would that work?

```
        $scope.getSelectionIndex = function() {
          for (var i = 0; i < $scope.options.length; i++) {
            if ($scope.options[i].val === $scope.selection) {
              return i;
            }
          }
        };
```